### PR TITLE
cinc-client: init at 18.2.7

### LINKED ISSUE #226773: Package request: cinc
**Project description**

Cinc is the Community build of  Chef. A systems integration framework, built to bring the benefits of configuration management to your entire infrastructure.

**Metadata**

* homepage URL: https://cinc.sh/
* source URL: https://github.com/cinc-project/chef
* license: Apache License 2.0
* platforms: unix

### PR DESCRIPTION
**Pull request by dylanmtaylor**
###### Description of changes

This adds the cinc-client application and closes #226773.

Regrettably, the gem is called chef-bin upstream.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/cinc-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{ lib, ruby, bundlerApp, bundlerUpdateScript }:
+
+bundlerApp {
+  pname = "chef-bin";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-client/default.nix:4]: This should match the directory name, cinc-client

File: pkgs/tools/misc/cinc-client/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  exes = ["cinc-client"];
+
+  passthru.updateScript = bundlerUpdateScript "chef-bin";
+
+  meta = with lib; {
+    description = "Cinc Client is the community build of the Chef infra client";
+    homepage = "https://cinc.sh/start/client/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ dylanmtaylor ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-client/default.nix:18]:
```suggestion
```

File: pkgs/tools/misc/cinc-client/default.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  gemdir = ./.;
+
+  inherit ruby;
+
+  exes = ["cinc-client"];
+
+  passthru.updateScript = bundlerUpdateScript "chef-bin";
+
+  meta = with lib; {
+    description = "Cinc Client is the community build of the Chef infra client";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-client/default.nix:14]: Why are we not packaging the chef infra client?

Comment (dylanmtaylor) [reply pkgs/tools/misc/cinc-client/default.nix:14]: That is a different PR. The difference is licensing. This is effectively a fork, like MyDQL -> MariaDB.

File: pkgs/tools/misc/cinc-client/default.nix (hunk 1 above)
Comment (dylanmtaylor) [reply pkgs/tools/misc/cinc-client/default.nix:4]: The gem for cinc-client is called chef-bin upstream, so the evaluation fails with it set in that manner. I don't know if there is an easy way to fix this.

File: pkgs/tools/misc/cinc-client/gemset.nix [hunk 4]
```diff
@@ -0,0 +1,1207 @@
...
+    };
+    version = "3.2.4";
+  };
+  chef = {
+    dependencies = ["addressable" "aws-sdk-s3" "aws-sdk-secretsmanager" "chef-config" "chef-utils" "chef-vault" "chef-zero" "corefoundation" "diff-lcs" "erubis" "ffi" "ffi-libarchive" "ffi-yajl" "iniparse" "inspec-core" "license-acceptance" "mixlib-archive" "mixlib-authentication" "mixlib-cli" "mixlib-log" "mixlib-shellout" "net-ftp" "net-sftp" "ohai" "plist" "proxifier2" "syslog-logger" "train-core" "train-rest" "train-winrm" "unf_ext" "uuidtools" "vault"];
+    groups = ["default"];
+    platforms = [];
+    source = {
+      remotes = ["https://packagecloud.io/cinc-project/stable"];
+      sha256 = "0yl42hjyid729a371p3xw0h4318adhx3rb39wvi6i6np8picircf";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cinc-client/gemset.nix:104]:
```suggestion
      sha256 = "sha256-4pGl4ZAtjcdRqdv4SQ6BvfYPP9b/Xgjhsun0LHyUo9s=";
```
are they doing ninja updates?

Comment (dylanmtaylor) [reply pkgs/tools/misc/cinc-client/gemset.nix:104]: That would be extremely unfortunate. It'd make version pinning hard.

**Comment by dylanmtaylor** [comment]
I am closing this as the Chef packages were merged in and I get the point that this is basically the same exact package but with a different name. The SHA checksum changing was also concerning from a maintenance perspective.

**Pull request closed without merging**